modules/plugins/utility/smart-splits/config.nix

```diff
@@ -0,0 +1,51 @@
+{
+  pkgs,
+  config,
+  options,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf;
+
+  cfg = config.vim.utility.smart-splits;
+  inherit (options.vim.utility.smart-splits) keymaps;
+  mkSmartSplitKey = act: let
+    key = cfg.keymaps.${act};
+  in
+    lib.optional (key != null) {
+      inherit key;
+      desc = keymaps.${act}.description;
+      action = ''function() require('smart-splits').${act}() end'';
+      mode = "n";
+      lua = true;
+    };
+in {
+  config = mkIf cfg.enable {
+    vim.lazy.plugins = {
+      ${pkgs.vimPlugins.smart-splits-nvim.pname} = {
```

Comment:
We don't use nixpkgs for plugins, please use it to npins instead.

Reply:
Of course, didn't know there was a preference. Will keep that in mind.